core/src/main/java/org/jboss/jandex/EmptyIndex.java

```diff
@@ -0,0 +1,88 @@
+package org.jboss.jandex;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Empty index; that is, an index that doesn't contain any class. All methods return either
+ * an empty collection, or {@code null}.
+ *
+ * @since 3.2.0
+ */
+public final class EmptyIndex implements IndexView {
```

Comment:
Ehm, what is this for? I mean it's not even used internally, is it? And what's the benefit of this optimized empty version?  Right now, there are other ways to obtain an empty index such as `Index.of(Collections.emptyList())`. Shouldn't we use the `EmptyIndex` in similar scenarios? Maybe we should add something like `Index.empty()`?

Reply:
So there's a difference between empty `Index` (which is fairly heavy-weight) and empty `IndexView` (which is super light-weight).

I don't have a use case for it myself, but apparently Hibernate Search people do: https://github.com/hibernate/hibernate-search/blob/main/util/common/src/main/java/org/hibernate/search/util/common/jar/impl/JandexUtils.java#L51 And it feels right to have a dedicated implementation directly in Jandex.

Reply:
If so, then we should probably promote the API a bit more, i.e. mention/link it in the `IndexView`, maybe even add `IndexView.empty()` etc.

Reply:
I did contemplate about a `public static IndexView empty()` method on `IndexView` vs. a `public class EmptyIndex` and didn't feel strongly about either. I can switch to the `static` method for sure.

Reply:
Added `IndexView.empty()`.

Reply:
I think that it's ok to have both.

Reply:
Yeah, agree, I didn't know why I didn't make `EmptyIndex.INSTANCE` public. Fixed.